metagraph/src/graph/alignment/aligner_aggregator.hpp

```diff
@@ -37,28 +39,88 @@ class AlignmentAggregator {
   private:
     const std::string_view query_;
     const std::string_view rc_query_;
+    const DeBruijnGraph &graph_;
     const DBGAlignerConfig &config_;
     boost::container::priority_deque<DBGAlignment,
                                      std::vector<DBGAlignment>,
                                      AlignmentCompare> path_queue_;
+    std::vector<score_t> best_score_;
+
+    void call_alignment_chains(const std::vector<DBGAlignment> &paths,
+                               const std::function<void(std::vector<DBGAlignment>&&,
+                                                        score_t)> &callback,
+                               size_t i = 0,
+                               std::vector<DBGAlignment>&& chain = {},
+                               score_t cur_score = 0,
+                               std::shared_ptr<std::vector<score_t>> best_score = {}) const;
 };
 
 
 template <typename NodeType, class AlignmentCompare>
 inline void AlignmentAggregator<NodeType, AlignmentCompare>
 ::add_alignment(DBGAlignment&& alignment) {
-    if (path_queue_.size() < config_.num_alternative_paths) {
-        path_queue_.emplace(std::move(alignment));
-    } else if (alignment.get_score() > path_queue_.minimum().get_score()) {
-        path_queue_.update(path_queue_.begin(), std::move(alignment));
+    if (!config_.chain_alignments) {
+        // no alignment chaining
+        if (path_queue_.size() < config_.num_alternative_paths) {
+            path_queue_.emplace(std::move(alignment));
+        } else if (alignment.get_score() > path_queue_.minimum().get_score()) {
+            path_queue_.update(path_queue_.begin(), std::move(alignment));
+        }
+    } else {
+        // alignment chaining
+        bool updated = false;
```

Comment:
to reduce the number of 'else' blocks...
```suggestion
    if (!config_.chain_alignments) {
        // no alignment chaining
        if (path_queue_.size() < config_.num_alternative_paths) {
            path_queue_.emplace(std::move(alignment));
        } else if (alignment.get_score() > path_queue_.minimum().get_score()) {
            path_queue_.update(path_queue_.begin(), std::move(alignment));
        }
        return;
    }

    // alignment chaining
    bool updated = false;
```